src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx

```diff
@@ -0,0 +1,183 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Box} from 'grid-emotion';
+
+import SelectControl from 'app/components/forms/selectControl';
+
+import {getInternal, getExternal, isValidCondition} from './utils';
+import {CONDITION_OPERATORS} from '../data';
+
+export default class Condition extends React.Component {
+  static propTypes = {
+    value: PropTypes.array,
+    onChange: PropTypes.func,
+    columns: PropTypes.array,
```

Comment:
Do columns have a type we can define with `PropTypes.shape` here?

Reply:
yep